lib/src/int64_native.dart

```diff
@@ -0,0 +1,376 @@
+        i = (i * radix) + digit;
+      } else {
+        if (!throwOnError) return null;
+        throw FormatException('Not radix digit', s, charIdx);
+      }
+    }
+
+    if (negative) {
+      return Int64(-i);
+    }
+
+    return Int64(i);
+  }
+
+  static int _promote(Object value) {
+    if (value is Int64) {
+      return value._i;
+    } else if (value is int) {
+      return value;
+    } else if (value is Int32) {
+      return value.toInt();
+    }
+    throw ArgumentError.value(value, 'other', 'not an int, Int32 or Int64');
+  }
+
+  @override
+  Int64 operator +(Object other) => Int64(_i + _promote(other));
+
+  @override
+  Int64 operator -(Object other) => Int64(_i - _promote(other));
+
+  @override
+  Int64 operator -() => Int64(-_i);
+
+  @override
+  Int64 operator *(Object other) => Int64(_i * _promote(other));
+
+  @override
+  Int64 operator %(Object other) => Int64(_i % _promote(other));
+
+  @override
+  Int64 operator ~/(Object other) => Int64(_i ~/ _promote(other));
+
+  @override
+  Int64 remainder(Object other) => Int64(_i.remainder(_promote(other)));
+
+  @override
+  Int64 operator &(Object other) => Int64(_i & _promote(other));
+
+  @override
+  Int64 operator |(Object other) => Int64(_i | _promote(other));
+
+  @override
+  Int64 operator ^(Object other) => Int64(_i ^ _promote(other));
+
+  @override
+  Int64 operator ~() => Int64(~_i);
+
+  @override
+  Int64 operator <<(int shiftAmount) => Int64(_i << shiftAmount);
+
+  @override
+  Int64 operator >>(int shiftAmount) => Int64(_i >> shiftAmount);
+
+  @override
+  Int64 shiftRightUnsigned(int shiftAmount) => Int64(_i >>> shiftAmount);
+
+  @override
+  int compareTo(Object other) => _i.compareTo(_promote(other));
+
+  @override
+  bool operator <(Object other) => _i < _promote(other);
+
+  @override
+  bool operator <=(Object other) => _i <= _promote(other);
+
+  @override
+  bool operator >(Object other) => _i > _promote(other);
+
+  @override
+  bool operator >=(Object other) => _i >= _promote(other);
+
+  @override
+  bool get isEven => _i.isEven;
+
+  @override
+  bool get isMaxValue => this == MAX_VALUE;
+
+  @override
+  bool get isMinValue => this == MIN_VALUE;
+
+  @override
+  bool get isNegative => _i < 0;
+
+  @override
+  bool get isOdd => _i.isOdd;
+
+  @override
+  bool get isZero => _i == 0;
+
+  @override
+  int get bitLength => _i.bitLength;
+
+  /// Returns a hash code based on all the bits of this [Int64].
+  @override
+  int get hashCode => _i.hashCode;
+
+  /// Returns [:true:] if this [Int64] has the same numeric value as the given
+  /// object.  The argument may be an [int] or an [IntX].
+  @override
+  bool operator ==(Object other) => _i == _promote(other);
+
+  @override
+  Int64 abs() => Int64(_i.abs());
+
+  @override
+  Int64 clamp(Object lowerLimit, Object upperLimit) =>
+      Int64(_i.clamp(_promote(lowerLimit), _promote(upperLimit)));
+
+  /// Returns the number of leading zeros in this [Int64] as an [int]
```

Comment:
The documentation on the abstract class says:
```
  /// Returns the number of high-order zeros in this integer's bit
  /// representation.
```
Is it any more clear than this one?

My suggestion is we remove this documentation, and in the abstract class we document the member as:
```
  /// Returns the number of leading (most significant) zeros in this integer's bit representation.
```

WDYT @lrhn?

Reply:
I think that final description is good. 
I'd still add a line saying what the result is for zero, just because it's a special case.

(In fact it's so special the result is probably useless. But then, I don't know what the use-case would be to begin with, and why it's not using `.bitLength` instead.
If you use it for padding, it's wrong: `n.toRadixString(2).padLeft(n.leadingZeros, '0')` gives a 65-digit numeral.
Maybe it should be 63 instead of 64? The same way `0.bitLength` is 1, not zero.
It's also the one case where `x.leadingZeros + x.trailingZeros` can be > 64.
What if it was consistent in that `x.leadingZeros + x.trailingZeros + (x >>> x.trailingZeros).bitLength == 64`
was always true, and for zero, we had `0.leadingZeros == 63` and `x.trailingZeros == 0`.) There are no zeros below the least significant bit. There is only one significant bit, and in this special case, it's zero. Otherwise the significant bits start and end with `1`.
Anyway, not doign a rewrite right here, just saying that `0.leadingZeros` is not given to be 64, so it should be documented. And probably deprecated at some point, if it's not really needed.)

Reply:
Done in #119, ptal.